tools/DeviceExplorer/DeviceExplorer/MainForm.cs

```diff
@@ -347,25 +350,34 @@ private void generateSASButton_Click(object sender, EventArgs e)
         #region ManagementTab
         private async Task updateDevicesGridView()
         {
-            var devicesProcessor = new DevicesProcessor(activeIoTHubConnectionString, MAX_COUNT_OF_DEVICES, protocolGatewayHost.Text);
-            var devicesList = await devicesProcessor.GetDevices();
-            devicesList.Sort();
-            allDevices = new SortableBindingList<DeviceEntity>(devicesList);
+            devicesGridView.ReadOnly = true;
+            devicesGridView.SelectionMode = DataGridViewSelectionMode.FullRowSelect;
 
-            string deviceCurrentlySelected = null;
+            devicesProcessor = new DevicesProcessor(
```

Comment:
User shouldn't be able to trigger several processors in parallel. How are we protecting against that?